torch/_dynamo/variables/user_defined.py

```diff
@@ -204,11 +206,12 @@ class UserDefinedObjectVariable(UserDefinedVariable):
     Mostly objects of defined type.  Catch-all for something where we only know the type.
     """
 
-    def __init__(self, value, value_type=None, **kwargs):
+    def __init__(self, value, value_type=None, type_tracker=None, **kwargs):
         super().__init__(**kwargs)
         self.value = value
         self.value_type = value_type or type(value)
         assert type(value) is self.value_type
+        self.type_tracker = type_tracker or variables.ConstantVariable(self.value_type)
```

Comment:
This seems kind of suspect, because usually a UDOV shouldn't get a ConstantVariable value_type? So now I have to audit all the constructor sites to tell if this has been done correctly.

What's wrong with using builder to recursively build the value_type in this case?

Reply:
This is the old behavior before this PR. (return constantvariable on type()). I'm leaving it because I don't know what will break and how much code changes I need to fix it completely.

Reply:
OK. Since you're still ramping onto the Dynamo codebase, I think it's worth following through with the rest of the sites to work out the implications. We can land this as is for now.